### PR TITLE
Move to GitHub actions

### PR DESCRIPTION
**Pull request by drvinceknight**

### CODE REVIEW

**Comment by drvinceknight** [comment]
This is ready to be merged now. 

All the work is really done in 3 commits:

- b940b39 (Subject: Add github actions and remove travis)
- dbe8a12 (Subject: Report coverage separately.)
- 5f08050 (Subject: Remove unnecessary badges.)

All the other commits are just testing that the CI reports failures correctly. If you look through the sequence of commits they all check that GitHub actions reports the required failure and then once the corresponding commit is reverted it passes. The red cross on a reverted commit is just because appveyor's PR build wasn't keeping up with my sequences of commits (as it was taking too long I didn't wait). You can check that all the builds behaved as expected by clicking on the red cross next to a commit.

For example, for 46bf990 (Subject: Revert "Add an error so docs do not build." ):

Screenshot 2020-04-07 at 16 40 40


**Appveyor**

To remove appveyor I need to do more than just delete `appveyor.yml`, I have to delete the project on the appveyor site. As there are other PRs going I won't do that until this PR is in. Once it is (and we have CI for windows and ubuntu through GitHub actions) then I'll delete the project on the appveyor site and open a PR to remove the `appveyor.yml`.

Once this is in I'll also open a few more issues with other things we can do (for example adding `py3.8` and an OSX run - I tried doing this briefly and small bugs appeared so we can just do it separately).

**The README**

You can view the README here: https://github.com/Axelrod-Python/Axelrod/blob/5f080503467c6c1158d2c54619445f8e3759e851/README.rst clicking on the CI badge takes you to the GitHub actions page.

@marcharper suggested we could have a 100% coverage sticker, but can we do that in a separate PR?

File: .github/workflows/config.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+jobs:
+  build:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      max-parallel: 4
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        python-version: [3.6, 3.7]
```
Comment (marcharper) [inline .github/workflows/config.yml:13]: Maybe just 3.7 and 3.8? Also we'll need to update the readme which currently says requires Python 3.5 or later (can do later). 3.5 may still work but we might as well say we support the latest two versions there.

Comment (drvinceknight) [reply .github/workflows/config.yml:13]: 100% but when I plugged 3.8 in some minor bugs cropped up so I'd like to take care of that in a separate PR if that's ok.

For now this PR gives us "as is" with travis + appveyor.

I'll update the README though :+1:

Comment (drvinceknight) [reply .github/workflows/config.yml:13]: a0bb6f3

**Comment by Nikoleta-v3** [comment]
Everything looks good to me as well! 

> Once this is in I'll also open a few more issues with other things we can do (for example adding py3.8 and an OSX run - I tried doing this briefly and small bugs appeared so we can just do it separately).

> @marcharper suggested we could have a 100% coverage sticker, but can we do that in a separate PR?

We should just remember to open these issues 👍

**Comment by drvinceknight** [comment]
f4bd6ab's failure is expected it's essentially checking that it caught the bug that https://github.com/Axelrod-Python/Axelrod/pull/1314 fixes which is its purpose.

**Pull request merged**